### PR TITLE
Only enable codecoverage on specific travis jobs

### PR DESCRIPTION
**Pull request by svenklemm**
Because add_link_options is not supported on older cmake versions
available in older postgresql docker images we only enable
code coverage on the codecov jobs.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -272,7 +272,8 @@ jobs:
     - if: (type = pull_request) OR (type = cron) OR NOT (branch = master)
       stage: test
       name: "Regression 9.6.17"
-      env: PG_VERSION=9.6.17
+      env:
+        - PG_VERSION=9.6.17
```
Comment (k-rus) [inline .travis.yml:276]: I don't follow why this change is needed. It makes it inconsistent with the jobs below.

Comment (svenklemm) [reply .travis.yml:276]: i'll remove i had multiple envs there for a while so this is leftover

**Comment by svenklemm** [comment]
> Since `-DCODECOVERAGE=ON` is removed from common script, which I guess is good to do. Should it be added to the CodeCov jobs?
> 
> Also minor comment about unnecessary change.

It is in OTHER_CMAKE_FLAGS for the codecov jobs already
